### PR TITLE
Add Lambda

### PR DESCRIPTION
**Pull request by gvcallen**
Adds the Lambda bijector. Implemented using distrax's [lambda](https://github.com/google-deepmind/distrax/blob/main/distrax/_src/bijectors/lambda_bijector.py) as a reference

### CODE REVIEW

**Comment by lockwo** [comment]
If we have Lambda do we need identity? https://github.com/google-deepmind/distrax/blob/main/distrax/_src/bijectors/lambda_bijector.py#L41

**Comment by gvcallen** [comment]
I do still think so, since its more about semantic clarity than anything else. For example Equinox itself has both [eqx.nn.Identity](https://docs.kidger.site/equinox/api/nn/linear/) and [eqx.nn.Lambda](https://docs.kidger.site/equinox/api/nn/sequential/).

File: distreqx/bijectors/_lambda.py [hunk 1]
```diff
@@ -0,0 +1,127 @@
+from collections.abc import Callable
+from typing import Optional
+
+import equinox as eqx
+from jaxtyping import PyTree
+
+# Assuming distreqx has ported these utilities from distrax
```
Comment (lockwo) [inline distreqx/bijectors/_lambda.py:7]: comment?

File: distreqx/utils/transformations.py [hunk 2]
```diff
@@ -0,0 +1,355 @@
...
+import warnings
+from collections.abc import Callable
+from typing import Any
+
+import jax
+import jax.extend as jex
+import jax.numpy as jnp
+from jaxtyping import Array, PyTree
+
+try:
```
Comment (lockwo) [inline distreqx/utils/transformations.py:28]: the try accept import isn't needed, we just have this as a requirement in jax versions

File: distreqx/utils/transformations.py [hunk 3]
```diff
@@ -0,0 +1,355 @@
+"""Utility functions for performing Bijector-related JAX transformations.
```
Comment (lockwo) [inline distreqx/utils/transformations.py:1]: this is from distrax? can reference source code

File: distreqx/utils/transformations.py [hunk 4]
```diff
@@ -0,0 +1,355 @@
...
+    - `fn`: The scalar function to derive the log-determinant for.
+
+    **Returns:**
+
+    - A callable that computes the log-determinant of the Jacobian of `fn`.
+    """
+    _check_numerical_stability(fn)
+    jac_fn = jax.vmap(jax.jacfwd(fn))
+
+    def log_det_fn(x: Array) -> Array:
```
Comment (lockwo) [inline distreqx/utils/transformations.py:169]: I'm not sure how well this will work with closures over eqx Modules. 

(Caution LLM output) I asked an LLM this question (rather than just testing cause I guess I'm lazy) and "For an eqx.Module implementing a * x, updating the stored forward callable’s a from 2 to 3 changes the forward output but leaves the derived log-Jacobian at log(2). I also reproduced a parameter gradient of 0 instead of 0.5 at a=2. This silently breaks density evaluation and training. Derived methods need to use the current callable and preserve its parameters in the pytree."

File: distreqx/utils/transformations.py [hunk 5]
```diff
@@ -0,0 +1,355 @@
...
+    jax.lax.erf_p: jax.lax.erf_inv_p,
+    jax.lax.erf_inv_p: jax.lax.erf_p,
+    jax.lax.conj_p: jax.lax.conj_p,
+    # binary ops; tuple values represent the variable-left/variable-right side
+    # case for non-commutatively invertible ops like div
+    jax.lax.mul_p: (jax.lax.div_p.bind, lambda x, y: jax.lax.div_p.bind(y, x)),
+    jax.lax.div_p: (jax.lax.mul_p.bind, jax.lax.div_p.bind),
+    jax.lax.add_p: (jax.lax.sub_p.bind, lambda x, y: jax.lax.sub_p.bind(y, x)),
+    jax.lax.sub_p: (jax.lax.add_p.bind, jax.lax.sub_p.bind),
+    jax.lax.pow_p: lambda x, y: jax.lax.pow_p.bind(x, 1.0 / y),
```
Comment (lockwo) [inline distreqx/utils/transformations.py:61]: this only handles a variable base, what about if variable is exponent?
